compiler/rustc_expand/src/mbe/macro_parser.rs

```diff
@@ -96,7 +96,8 @@ use std::collections::hash_map::Entry::{Occupied, Vacant};
 ///
 /// This means a matcher can be represented by `&[MatcherLoc]`, and traversal mostly involves
 /// simply incrementing the current matcher position index by one.
-pub(super) enum MatcherLoc {
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum MatcherLoc {
```

Comment:
Why are `Clone` and `PartialEq` needed?

Reply:
I think that's an artifact of earlier states of this, I should be able to remove it.